demos/esc/README.md

```diff
@@ -0,0 +1,85 @@
+# ESC Library Demo - eurus-project
+
+This is the example application demonstrating usage of the 
+Eurus Flight Controller (EFC) ESC (Electronic Speed Controller)
+library within the Zephyr framework.
+
+## Features & Demo configuration
+
+The library supports multiple ESC protocols, including:
+- Classic PWM
+- Oneshot 125
+- Oneshot 42
+- Multishot
+
+Default ESC protocol in this sample is PWM, but it can be reconfigured 
+using menuconfig. To start menuconfig, the demo needs to be built with: 
+
+`west build -b <board_name> <path_to_demo>`
+
+then, to start menuconfig enter:
+
+`west build -t menuconfig`
+
+![Eurus Flight Controller - ESC Demo menuconfig](.docs/menuconfig1.png)
+
+Next, by choosing ESC Protocol Selection, submenu will open where ESC PWM
+protocol has been choosen by default. 
+
+![ESC Protocol Selection submenu](.docs/menuconfig2.png)
+
+Any other protocol can be choosen (i.e. Oneshot42) by moving ↑ and ↓ + Enter
+
+![ESC Protocol Selection - Oneshot42](.docs/menuconfig3.png)
+
+When that configuration has been set, ESC library must be set and configured:
+
+![ESC Library settings](.docs/menuconfig4.png)
+
+All analog ESC protocols have a standard minimal and maximal pulse durations:
+- PWM        -> 1000-2000 us
+- Oneshot125 -> 125-250 us
+- Oneshot42  -> 42-84 us
+- Multishot  -> 5-25 us
+
+But in reality, this values can vary from vendor to vendor. In this demo, ESC 
+that has been used is BLHeliS ESC, and it has a dead zone of 30 % starting from
+a minimal pulse duration. (Motor starts spining only when 30 percent of speed 
+has been reached.) By using **Enable manual minimum thrust offset for ESC** option,
+this offset value can be configured.
+
+![Enable manual minimum thrust offset for ESC](.docs/menuconfig5.png)
+
+**Enable manual maximum thrust offset for ESC** can be used to set offset of maximum pulse duration, but in this demo it wasn't necessary.
+
+## Demo
+
+### Requirements
+
+#### Hardware
```

Comment:
Nitpick: There is no need to have this many levels of headers, the same effect can be achieved with just one or two.